config/main.py

```diff
@@ -1227,6 +1227,92 @@ def validate_patch(patch):
     except Exception as e:
         raise GenericConfigUpdaterError(f"Validate json patch: {patch} failed due to:{e}")
 
+
+def multiasic_validate_single_file(filename):
+    ns_list = [DEFAULT_NAMESPACE, *multi_asic.get_namespace_list()]
+    file_input = read_json_file(filename)
+    file_ns_list = [DEFAULT_NAMESPACE if key == HOST_NAMESPACE else key for key in file_input]
+    if set(ns_list) != set(file_ns_list):
+        diff = set(ns_list) - set(file_ns_list)
+        click.echo("Input file {} must contain all asics config. Missing {}".format(filename, diff))
+        raise click.Abort()
+
+
+def load_sysinfo_if_missing(asic_config):
+    platform = asic_config.get("DEVICE_METADATA", {}).\
+        get(HOST_NAMESPACE, {}).get("platform")
+    mac = asic_config.get("DEVICE_METADATA", {}).\
+        get(HOST_NAMESPACE, {}).get("mac")
+    if not platform or not mac:
+        log.log_warning("Input file does't have platform or mac. platform: {}, mac: {}"
+                        .format(None if platform is None else platform, None if mac is None else mac))
```

Comment:
>if [](http://example.com/codeflow?start=37&length=2)

If you just want to print error message, keep statements simple, for example:
1. if mac is missing, warn that mac is missing
2. if platform is missing, warm that platform is missing